doc/deployment/nginx_and_url_path_config.rst

```diff
@@ -0,0 +1,137 @@
+NGINX and URL Path Configuration for Eventyay-Tickets
+=====================================================
+
+This document outlines the NGINX and URL path setup for the ``eventyay-tickets`` project hosted at `https://github.com/fossasia/eventyay-tickets`. This Django project utilizes ``FORCE_SCRIPT_NAME`` to configure a base path, allowing the application to run on a specified sub-path within the domain.
+
+Project Setup in Django
+-----------------------
+
+In ``eventyay-tickets``, the following settings are used to configure the base path for the application:
+
+.. code-block:: python
+
+    # Django Settings
+    BASE_PATH = config.get('pretix', 'base_path', fallback='/tickets')
+    FORCE_SCRIPT_NAME = BASE_PATH
+
+- **BASE_PATH**: Retrieved from the ``pretix.cfg`` configuration file, which allows flexible configuration of the base URL path. By default, this is set to ``'/tickets'``.
+- **FORCE_SCRIPT_NAME**: Uses the value from ``BASE_PATH``, setting ``/tickets`` as the base path for URL routing within Django.
+
+This means that all URLs within Django are prefixed with ``/tickets``, allowing users to access ``eventyay-tickets`` via URLs like ``domain.com/tickets/``.
+
+NGINX Configuration
+-------------------
+
+The following NGINX configuration serves the ``eventyay-tickets`` application along with other services such as ``talk`` and ``video``. The configuration is hosted on ``app-test.eventyay.com``.
+
+Key Sections of NGINX Configuration
+-----------------------------------
+
+1. **Server Block for HTTPS**:
+   This block listens on port ``443`` for HTTPS traffic.
+   SSL certificates are configured here using Let's Encrypt for secure communication.
+
+   .. code-block:: nginx
+
+       server {
+           server_name app-test.eventyay.com;
+           listen 443 ssl;
+
+           ssl_certificate /etc/../fullchain.pem;
+           ssl_certificate_key /etc/../privkey.pem;
+           include /etc/../options-ssl-nginx.conf;
+           ssl_dhparam /etc/../ssl-dhparams.pem;
+       }
+
+2. **URL Path Configurations**:
+
+   - **Root Location**:
+     - Proxies requests from ``https://app-test.eventyay.com/`` to the backend on ``localhost:8455/common/``.
+
+   - **/tickets**:
+     - This location is for the ``eventyay-tickets`` application.
+     - Requests to ``https://app-test.eventyay.com/tickets/`` are proxied to ``localhost:8455/``.
```

Comment:
The documentation should be URL agnostic. If we deploy this somewhere else it does not make sense.